lib/Sema/CodeSynthesis.cpp

```diff
@@ -2042,6 +2042,12 @@ void swift::maybeAddAccessorsToStorage(TypeChecker &TC,
     return;
 
   if (!dc->isTypeContext()) {
+    // dynamic globals need accessors.
+    if (dc->isModuleScopeContext() && storage->isDynamic() &&
```

Comment:
`isDynamic() && !isObjC()` seems like a common enough predicate in your changes that it might be worth extracting isObjCDynamic() and isNativeDynamic().